File: Projects/Cpp/Game/Breakout/BreakoutBrick.cpp
#include "BreakoutBrick.hpp"
#include "AssetManager.h"

BreakoutBrick::BreakoutBrick(const Vector2 position)
{
    collisionRectangle.x = int(position.x);
    collisionRectangle.y = int(position.y);
    color = Color{u_char(GetRandomValue(0, 255)), u_char(GetRandomValue(0, 255)), u_char(GetRandomValue(0, 255)), 255};
}

void BreakoutBrick::render()
{
    DrawTexturePro(AssetManager::getInstance().getTexture("brick"),
                   AssetManager::getInstance().getRectangle("brick"),
                   collisionRectangle,
                   Vector2{0, 0},
                   0,
                   color);
//    DrawRectanglePro(collisionRectangle, Vector2{}, 0, ORANGE);
}

Rectangle BreakoutBrick::getCollisionRectangle()
{
    return collisionRectangle;
}

void BreakoutBrick::destroy()
{
    _shouldDestroy = true;
}

bool BreakoutBrick::shouldDestroy()
{
    return _shouldDestroy;
}

File: Projects/Java/Breakout/core/src/com/lukahorvat/breakout/BreakoutBrick.java
package com.lukahorvat.breakout;

import com.badlogic.gdx.assets.AssetManager;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.MathUtils;
import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.math.Vector2;

public class BreakoutBrick {
    private Rectangle collisionRectangle = new Rectangle(0, 0, 80, 30);
    private Color color;
    private boolean _shouldDestroy = false;

    public BreakoutBrick(Vector2 position)
    {
        collisionRectangle.x = position.x;
        collisionRectangle.y = position.y;
        color = new Color(MathUtils.random(),MathUtils.random(),MathUtils.random(),1);
    }

    public void render(SpriteBatch batch, AssetManager manager) {
        batch.setColor(color);
        batch.draw(manager.get("brick.png", Texture.class), collisionRectangle.x, collisionRectangle.y, collisionRectangle.width, collisionRectangle.height);
        batch.setColor(Color.WHITE);
    }

    public Rectangle getCollisionRectangle() {
        return collisionRectangle;
    }

    public void destroy() {
        _shouldDestroy = true;
    }

    public boolean shouldDestroy()
    {
        return _shouldDestroy;
    }
}
